Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1287,9 +1305,25 @@ class CombinatorialKalmanFilter {
     auto propState =
         m_propagator.template makeState(initialParameters, propOptions);
 
-    auto& r = propState.template get<CombinatorialKalmanFilterResult<traj_t>>();
-    r.fittedStates = &trackContainer.trackStateContainer();
-    r.stateBuffer = std::make_shared<traj_t>();
+    auto& r =
+        propState
+            .template get<CombinatorialKalmanFilterResult<track_container_t>>();
+    r.tracks = &trackContainer;
+    r.trackStates = &trackContainer.trackStateContainer();
+    r.stateBuffer = std::make_shared<
+        typename track_container_t::TrackStateBackendContainer>();
+
+    if (!trackContainer.hasColumn(tipStateAccessor.key)) {
+      ACTS_ERROR(
+          "Track container does not have the tip state column \"CkfTipState\"");
```

Comment:
Just checking... I guess this is an `ERROR` because it probably should have been done earlier.

Reply:
Right I think we should communicate this somehow to the user otherwise the component access will throw an exception / signal